packages/injective-cosmwasm/src/types.rs

```diff
@@ -149,6 +149,114 @@ impl<'a> Prefixer<'a> for &'a SubaccountId {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, JsonSchema)]
+pub struct TFDenom(String); // format!("factory/{}/lp{}", master_address, vault_address)
+
+impl TFDenom {
+    pub fn new<S>(denom: S) -> StdResult<Self>
+    where
+        S: Into<String>,
+    {
+        let denom = denom.into();
+
+        if !denom.starts_with("factory/") {
+            return Err(StdError::generic_err(
+                "Invalid prefix: token factory denoms must start with \"factory/...\"",
+            ));
+        }
+
+        if denom.len() != "factory/".len() + 33 + "/lp".len() + 33 {
```

Comment:
sure about that `33` length for subdenom? On chain we limit it to `44`.

Reply:
this is for a bech32 addresses (master+vault) or not? which actually are `42`, so I'm confused, but this is not the right place to have it anyways, see main review comment